coordinator-tool/src/bin/ec.rs

```diff
@@ -0,0 +1,895 @@
+    let (sender, mut receiver) = tokio::sync::mpsc::channel(20);
+    let cancel_handle = tokio::spawn(traverse_config.traverse(
+        indexer::ContractUpdateIndexer {
+            target_address,
+            entrypoint: OwnedEntrypointName::new_unchecked("registerVotes".into()),
+        },
+        sender,
+    ));
+
+    let bar = ProgressBar::new(last_block.block_height.height - first_block.block_height.height)
+        .with_style(ProgressStyle::with_template(
+            "{spinner} {msg} {wide_bar} {pos}/{len}",
+        )?);
+
+    let mut ballots = BTreeMap::new();
+
+    while let Some((block, txs)) = receiver.recv().await {
+        bar.set_message(block.block_slot_time.to_string());
+        bar.inc(1);
+        if block.block_slot_time > end {
+            drop(receiver);
+            cancel_handle.abort();
+            drop(cancel_handle);
+            eprintln!("Done indexing.");
+            break;
+        }
+
+        for indexer::ContractUpdateInfo {
+            execution_tree,
+            transaction_hash,
+            sender,
+            ..
+        } in txs
+        {
+            let param = execution_tree.parameter();
+            let Ok(param) = concordium_std::from_bytes::<contract::RegisterVotesParameter>(param.as_ref()) else {
+                eprintln!("Unable to parse ballot from transaction {transaction_hash}");
+                continue;
+            };
+
+            let Ok(ballot) = rmp_serde::from_slice::<eg::ballot::BallotEncrypted>(param.as_ref()) else {
+                eprintln!("Unable to parse ballot from transaction {transaction_hash}");
+                continue;
+            };
+            let verified = ballot.verify(
+                &verification_context,
+                eg::index::Index::from_one_based_index(1).unwrap(),
+            );
+            if verified {
+                // Replace any previous ballot from the sender.
+                ballots.insert(AccountAddressEq::from(sender), (ballot, transaction_hash));
+            } else {
+                eprintln!("Vote in transaction {transaction_hash} is invalid.");
+            }
+        }
+    }
+
+    let mut final_weights =
+        csv::Reader::from_path(final_weights).context("Unable to open final weights file.")?;
+
+    let mut tally = eg::ballot::BallotTallyBuilder::new(&election_data.manifest);
+    for row in final_weights.deserialize() {
+        let FinalWeightRow {
+            account,
+            amount,
+            delegators,
+        } = row?;
+        if let Some((ballot, hash)) = ballots.remove(&AccountAddressEq::from(account)) {
+            let factor = amount.micro_ccd() / 1_000_000u64;
+            eprintln!(
+                "Scaling the ballot cast by transaction {hash} by a factor {factor}. Delegators \
+                 {delegators}."
+            );
+            // TODO: After benchmarks are complete adjust the factor
```

Comment:
Have they completed? Otherwise, do we have an issue to track this?

Reply:
Yes, they have (this is the discussion about using CCD as weight).

I already adjusted the scale, but apparently forgot the TODO :)